src/commands/commerce/extension/map.ts

```diff
@@ -147,4 +147,33 @@ export class MapExtension extends SfdxCommand {
             return returnResult;
         }
     }
+    private deleteduplicateMaps(extensionName: string, userName: string): void {
+        try {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+            const EPNQuery = forceDataSoql(
+                `SELECT ExtensionPointName FROM RegisteredExternalService WHERE DeveloperName='${extensionName}'`,
+                userName
+            ).result.records[0]['ExtensionPointName'];
+            const existingIds = forceDataSoql(
+                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                `SELECT DeveloperName,ExternalServiceProviderType FROM RegisteredExternalService WHERE ExtensionPointName='${EPNQuery}' AND ExternalServiceProviderType='Extension'`,
+                userName
+            );
+            for (const element of existingIds.result.records) {
+                const id = (element['ExternalServiceProviderType'] as string)
+                    .concat('__' as string)
+                    .concat(element['DeveloperName'] as string);
+                try {
+                    const deleteId = forceDataSoql(
+                        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                        `SELECT Id FROM StoreIntegratedService WHERE Integration='${id}'`,
+                        userName
+                    ).result.records[0].Id;
+                    forceDataRecordDelete('StoreIntegratedService', deleteId, this.org.getUsername());
+                    // eslint-disable-next-line no-empty
+                } catch (e) {}
+            }
+            // eslint-disable-next-line no-empty
+        } catch (e) {}
```

Comment:
these catch blocks don't seem to handle any exceptions would we consider just removing them so that exceptions are thrown all the way up?

Reply:
@hajinsuha1 I kept the inner catch empty, so if a record exists in registeredExternalService, but not in StoreIntegratedService, it will still iterate to the next element. For example if test12,test13,test14 with EPN testEPN exists in registeredExternal, but only test14 exists in StoreIntegrated, then it will throw an error when trying to delete test12,test13 and will never get to the deletion of test14.

Reply:
I see then instead of trying to delete the error, would we consider checking if deleteId is not null before deleting? That way it won't try to delete an unmapped extension